ctest targets from catch2 and fastcov (#761)
STORM-IRIT/Radium-Engine: doc/developer/develWithRadium.md

Here is an example:

\snippet unittest/Core/obb.cpp obb test

# Code coverage

> On PR, code coverage analysis is perform through a github action with help of codecov.io report.

Reviewer: ```suggestion
On PR, code coverage analysis is performed through a github action with help of codecov.io report.
```

You can run such analysis on your own (on linux system, not tested on other plateform) using lcov.

> First install `lcov`, and ` libxml2-utils` for the tests.

Author: replaced by lcov, xmllint and sed.

It's propably on your system repositories, for Debian for instance `sudo apt install lcov libxml2-utils`.

Then build Radium with coverage enable, run test and generate report.

```bash
#some vars

> RADIUMC_CONFIG=path/to/file/with/radium-options.cmake #lik where external are built ... it's optionnal

Author: ```suggestion
RADIUM_CONFIG=path/to/file/with/radium-options.cmake #like where external are built ... it's optionnal
```

RADIUM_SOURCE=path/to/radium/main/CMakeLists.txt

# in you build path

> cmake -DCMAKE_BUILD_TYPE=Debug -DRADIUM_ENABLE_COVERAGE=ON -C RADIUM_CONFIG RADIUM_SOURCE

Author: ```suggestion
cmake -DCMAKE_BUILD_TYPE=Debug -DRADIUM_ENABLE_COVERAGE=ON -C ${RADIUM_CONFIG} ${RADIUM_SOURCE}
```

make lcov #build everything, with coverage enable, run test, perform analysis
make lcov-report # generate html report
```
output report is `./lcov/index.html`

`lcov` is quite slow to process files, [`fastcov`](https://github.com/RPGillespie6/fastcov/blob/master/fastcov.py) is a parallel alternative.
To install fastcov `pip3 install fastcov`

> and for the analysiss

Reviewer 2: ```suggestion
and for the analysis
```
